### PR TITLE
Made AbstractFixerTestCase and AbstractIntegrationTestCase public

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

**Comment by keradus** [comment]
Fabbot failed as usual...

**Comment by keradus** [comment]
any thoughts ?

**Comment by keradus** [comment]
rebased

**Comment by Soullivaneuh** [comment]
A note about BC break should be added on the upgrade gad IMO.

Althoug the class was marked as internal, it's still usable and probably use on 1.x.

**Comment by keradus** [comment]
I don't get it. Yes, the classes were internal, now they are not - how could it be a BC break?
Changing internal interface isn't a BC break.

Also, we have stable 1.x, and this PR is for 2.x, so even if it was a BC breaker, we are already targetting branch with major version change.

**Comment by Soullivaneuh** [comment]
> Yes, the classes were internal, now they are not - how could it be a BC break?

The class is renamed also. And even if it's marked as internal, it could be technically be used actually.

This should not, I agree, but it could AFAIK.

> Also, we have stable 1.x, and this PR is for 2.x, so even if it was a BC breaker, we are already targetting branch with major version change.

Yes, I was not talking about removing or handling BC break. Just put a note to `UPGRADE-2.0.md` file. :+1:

**Comment by keradus** [comment]
> The class is renamed also. And even if it's marked as internal, it could be technically be used actually.
> This should not, I agree, but it could AFAIK.

If one decided to use internal code he should not be surprised when it changed. We do not promise to keep the compatibility of it, then it is not a BC break. No exceptions.


> Yes, I was not talking about removing or handling BC break. Just put a note to `UPGRADE-2.0.md` file. :+1:

Please, feel free to send a PR with that file, I will be more than happy to see it ;)

File: Symfony/CS/Test/AbstractIntegrationTestCase.php [hunk 1]
```diff
@@ -137,7 +135,7 @@ protected static function getTempFile()
      * @param string           $input        Code to fix
      * @param string|null      $expected     Expected result or null if the input is expected not to change
      */
-    protected function doTestIntegration($testFileName, $testTitle, $fixers, $input, $expected = null)
+    protected function doTest($testFileName, $testTitle, $fixers, $input, $expected = null)
```
Comment (stof) [inline Symfony/CS/Test/AbstractIntegrationTestCase.php]: does it need to be protected here ?

**Comment by keradus** [comment]
Whole `Tests` dir could be moved, but it is a work for separate PR.

File: Symfony/CS/Test/AbstractIntegrationTestCase.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Test/AbstractIntegrationTestCase.php]: maybe not needed, but one may like to override it, adding some extra checks

**Comment by keradus** [comment]
Rebased, then I think it is ready to merge

**Pull request closed without merging**